include/aikido/perception/PerceptionModule.hpp

```diff
@@ -26,13 +27,15 @@ class PerceptionModule
   /// \param[in] timestamp Only detections more recent than this timestamp will
   /// be accepted. A timestamp of 0 greedily takes the first available message,
   /// and is the default behaviour.
+  /// \param[out] detectedObjects An output vector for detected objects.
   /// \return Returns \c false if no detection observed, or if none of the
```

Comment:
Hmm. I think this return parameter is misleading and is redundant. If there is no detection available since `timestamp`, the `detectedObjects` should be empty.

Reply:
It is useful for two reasons:
(1) If the user doesn't use detectedObjects (passes in `nullptr` / keeps to the default)
(2) If a user wants to do quick error detection without accessing detectedObjects. e.g.
```
while(!detectedObjects()) {
printf("Waiting for objects!);
}
```